### ISSUE TITLE
nameref assignments misunderstood

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1543]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

_ekvstore_nonatomic_set () {
  declare -n ns="$1"
  local key="$2"
  local val="$3"

  ns[$key]=$val
}


```

#### Here's what shellcheck currently says:
```
Line 8	SC2034: ns appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

Nothing. `declare -n` creates an alias to the variable whose name is in $1. Setting it should be considered "usage" in this case.

**Comment by jostockley** [comment]
Same applies when using local instead of declare:
#!/bin/bash

_ekvstore_nonatomic_set () {
  **local** -n ns="$1"
  local key="$2"
  local val="$3"

  ns[$key]=$val
}

**Comment by pabs3** [comment]
This appears to be a duplicate of #817.
